File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Translation.java
package io.getstream.chat.docs.java;

import io.getstream.chat.android.client.ChatClient;
import io.getstream.chat.android.client.channel.ChannelClient;
import io.getstream.chat.android.client.models.Message;

public class Translation {
    private ChatClient client;

    /**
     * @see <a href="https://getstream.io/chat/docs/translation/?language=java#message-translation-endpoint">Message Translation</a>
     */
    public void messageTranslation() {
        // Translate message to French
        ChannelClient channelClient = client.channel("messaging", "general");

        Message message = new Message();
        message.setText("Hello, I would like to have more information about your product.");

        channelClient.sendMessage(message).enqueue(result -> {
            if (result.isSuccess()) {
                String messageId = result.data().getId();

                client.translate(messageId, "fr").enqueue(translationResult -> {
                    if (translationResult.isSuccess()) {
                        Message translatedMessage = translationResult.data();
                        String translation = translatedMessage.getI18n().get("fr_text");
                    } else {
                        // Handle translationResult.error()
                    }
                });
            } else {
                // Handle result.error()
            }
        });
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Translation.kt
package io.getstream.chat.docs.kotlin

import io.getstream.chat.android.client.ChatClient
import io.getstream.chat.android.client.models.Message
import io.getstream.chat.android.client.models.getTranslation

class Translation(val client: ChatClient) {

    /**
     * @see <a href="https://getstream.io/chat/docs/translation/?language=kotlin#message-translation-endpoint">Message Translation</a>
     */
    fun messageTranslation() {
        // Translate message to French
        val channelClient = client.channel("messaging", "general")
        val message = Message(text = "Hello, I would like to have more information about your product.")

        channelClient.sendMessage(message).enqueue { result ->
            if (result.isSuccess) {
                val messageId = result.data().id
                val frenchLanguage = "fr"

                client.translate(messageId, frenchLanguage).enqueue { translationResult ->
                    if (translationResult.isSuccess) {
                        val translatedMessage = translationResult.data()
                        val translation = translatedMessage.getTranslation(frenchLanguage)
                    } else {
                        // Handle translationResult.error()
                    }
                }
            } else {
                // Handle result.error()
            }
        }
    }
}
